### ISSUE TITLE
How to interpret the results of Kstar?

### ISSUE THREAD

**Issue by biotech70** [donaldlab/OSPREY3#164]
I implemented the Kstar algorithm for desired complex, but I do not know how to interpret the results because this tool has not any tutorial.
1- How do we recognize the best mutation? 
2- Why is there no output for some mutations and what does Infinity mean here?
3- how interpret these lines and what do they mean?
![2](https://user-images.githubusercontent.com/73783725/116884469-03668e00-ac3c-11eb-9cd7-a79beb9b982b.png)
![k_score](https://user-images.githubusercontent.com/73783725/116884485-06617e80-ac3c-11eb-8ee4-a2b791603ffa.png)
![3](https://user-images.githubusercontent.com/73783725/116884511-0eb9b980-ac3c-11eb-922a-43db58356cc8.png)

**Comment by cuchaz** [comment]
Hi there, I'll do my best to answer your questions. Sorry, Osprey currently doesn't have a design manual because no one has written one yet.

> 1- How do we recognize the best mutation?

The K* score is an approximation to a binding constant, so a higher K* score predicts a tighter binder.
The [original K* paper](https://users.cs.duke.edu/~brd/papers/src-papers/pdf-reprints/jcb05-nrps-for-printing.pdf) may help you learn more.

> 2- Why is there no output for some mutations and what does Infinity mean here?

The K* score is a ratio of partition function values. If a value in the denominator is zero, for example, that can result in an infinite K* score. If a partition function value is zero, usually that means Osprey didn't have enough precision to calculate the approximation. Usually that's caused by conformations in that conformation having relatively high energies, resulting in Boltzmann weights *very* close to zero.

> 3- how interpret these lines and what do they mean?

Those lines define the binding partners for your design from portions of the PDB file. The `residues` argument gives the range of residue names from the PDB file.

> in last, score or energy which one is important and what is their relationship?

You really only need to care about the energy. The energy is the forcefield-calculated energy of the conformation. The score is just an internal implementation detail. It's a predicted lower bound on the energy and it's just used by the conformation enumeration algorithm to find candidate conformations that might have low energies. As you can see from the results you posted, the score isn't always well-correlated with the energy, so the conformation search algorithm has to do a lot of work!

**Comment by biotech70** [comment]
Thanks for your comprehensive answer

**Comment by biotech70** [comment]
I try to compute binding energy for spike-ACE2 complex following N501Y mutation. Although in many investigations its reported that this mutation increase binding affinity of spike for binding to ACE2, OSPREY report infinity for this mutation.

**Comment by cuchaz** [comment]
If Osprey reports infinity for a sequence, usually that means there's a clash somewhere. If you can find that clash and fix it, you might get Osprey to report a more useful answer next time. Maybe try looking at the lowest-energy structure for that sequence and see if something looks wrong?

**Comment by biotech70** [comment]
I used from ROSETTA docking output for OSPREY and all clashes are fixed by refine and relax module of ROSETTA, but got error.

**Comment by cuchaz** [comment]
That's the forcefield energy of a complex structure. That display just shows forcefield energies for the complex state.

**Comment by biotech70** [comment]
Wow, I thought it was binding energy.

**Comment by biotech70** [comment]
Of course, I think you are wrong, and the reported energy is the same as the binding energy. See the following articles:
1-OSPREY Predicts Resistance Mutations using Positive and Negative Computational Protein Design
2-OSPREY 3.0: Open-Source Protein Redesign for You, with Powerful New Features

**Comment by biotech70** [comment]
How to fix the following error? kstar.py exists on desired directory but i get error
![Untitled](https://user-images.githubusercontent.com/73783725/117583364-02da6580-b11c-11eb-9c43-7c68704da764.png)

**Comment by cuchaz** [comment]
Hmm... looks like your Osprey installation got mixed up somehow. I'll respond to your other issue and maybe that will fix the problem.

**Comment by biotech70** [comment]
Thank you for your patience and response.

**Comment by biotech70** [comment]
I am a biologist and I am very interested in working with Dr Ivelin Georgiev, who has been involved in the development of OSPREY, because he is currently working on antibodies.

**Comment by cuchaz** [comment]
Oh, cool! Ivelin wrote a lot of the early code for K*. I had the privilege of working with him briefly in the Donald lab during our PhDs.
